relays/ethereum/src/rpc_errors.rs

```diff
@@ -101,20 +102,34 @@ pub enum EthereumNodeError {
 	IncompleteHeader,
 	/// We have received a receipt missing a `gas_used` field.
 	IncompleteReceipt,
+	/// We have received a transaction missing a `raw` field.
+	IncompleteTransaction,
 	/// An invalid Substrate block number was received from
 	/// an Ethereum node.
 	InvalidSubstrateBlockNumber,
+	/// Block includes the same transaction more than once.
+	DuplicateBlockTransaction(EthereumHeaderId, EthereumTransactionHash),
+	/// Block is missing transaction we believe is a part of this block.
+	BlockMissingTransaction(EthereumHeaderId, EthereumTransactionHash),
 }
 
 impl ToString for EthereumNodeError {
 	fn to_string(&self) -> String {
 		match self {
-			Self::ResponseParseFailed(e) => e,
-			Self::IncompleteHeader => "Incomplete Ethereum Header Received",
-			Self::IncompleteReceipt => "Incomplete Ethereum Receipt Recieved",
-			Self::InvalidSubstrateBlockNumber => "Received an invalid Substrate block from Ethereum Node",
+			Self::ResponseParseFailed(e) => e.to_string(),
+			Self::IncompleteHeader => "Incomplete Ethereum Header Received".to_string(),
```

Comment:
Could you be more explicit about what's missing in the header? This would help the users debug errors.